core/controllers/editor.py

```diff
@@ -56,15 +54,42 @@ def _require_valid_version(version_from_payload, exploration_version):
             % (exploration_version, version_from_payload))
 
 
+# Common schemas used in this file.
+SCHEMA_FOR_EXPLORATION_ID = {
+    'type': 'basestring',
+    'validators': [{
+        'id': 'is_regex_matched',
+        'regex_pattern': r'^[a-zA-Z0-9\-_]{1,12}$'
```

Comment:
This regex_pattern will be present in feconf after merge of #13456. 
Willl update after it, thank you